algorithms/linfa-clustering/src/k_means/algorithm.rs

```diff
@@ -444,6 +444,31 @@ impl<F: Float, DA: Data<Elem = F>, D: Distance<F>> PredictRef<ArrayBase<DA, Ix1>
     }
 }
 
+impl<F: Float, DA: Data<Elem = F>, D: Distance<F>> PredictInto<ArrayBase<DA, Ix2>, Array1<usize>>
+    for KMeans<F, D>
+{
+    /// Predict into a mutable reference of `targets`
+    ///
+    /// # Panics
+    /// `targets` must be sufficiently large to hold the targets `observations.nrows() <= targets.len()`
+    fn predict_into<'a>(
+        &'a self,
+        observations: &'a ArrayBase<DA, Ix2>,
+        targets: &mut Array1<usize>,
+    ) {
+        assert!(
+            observations.nrows() <= targets.len(),
```

Comment:
I think you should panic on inputs that are too long as well, since that's indicative of a programming error.

Reply:
I was thinking that it is more flexible that way but an equality check should avoid more errors.